File: ehe/ehe.Web/Modules/DosyaUrunler/UrunCesitleri/UrunCesitleriRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace ehe.DosyaUrunler;

[ConnectionKey("Default"), Module("DosyaUrunler"), TableName("UrunCesitleri")]
[DisplayName("Urun Cesitleri"), InstanceName("Urun Cesitleri")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]

[LookupScript]

public sealed class UrunCesitleriRow : Row<UrunCesitleriRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("id"), Identity, IdProperty]
    public int? UrunCesitleriid { get => fields.UrunCesitleriid[this]; set => fields.UrunCesitleriid[this] = value; }

    [DisplayName("Ad"), Size(100), NotNull, QuickSearch, NameProperty]
    public string Ad { get => fields.Ad[this]; set => fields.Ad[this] = value; }

    [DisplayName("Kısa Ad"), Size(3), NotNull, LookupInclude]
    public string ShortName { get => fields.ShortName[this]; set => fields.ShortName[this] = value; }


    public class RowFields : RowFieldsBase
    {
        public Int32Field UrunCesitleriid;
        public StringField Ad;
        public StringField ShortName;

    }
}

File: ehe/ehe.Web/Modules/ServerTypes/DosyaUrunler/UrunCesitleriRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface UrunCesitleriRow {
    UrunCesitleriid?: number;
    Ad?: string;
    ShortName?: string;
}

export abstract class UrunCesitleriRow {
    static readonly idProperty = 'UrunCesitleriid';
    static readonly nameProperty = 'Ad';
    static readonly localTextPrefix = 'DosyaUrunler.UrunCesitleri';
    static readonly lookupKey = 'DosyaUrunler.UrunCesitleri';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<UrunCesitleriRow>('DosyaUrunler.UrunCesitleri') }
    static async getLookupAsync() { return getLookupAsync<UrunCesitleriRow>('DosyaUrunler.UrunCesitleri') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<UrunCesitleriRow>();
}